apps/condo/domains/banking/components/BankAccountVisibilitySelect.tsx

```diff
@@ -35,22 +38,23 @@ export const BankAccountVisibilitySelect: IBankAccountVisibilitySelect = ({ bank
     const accountVisibility = get(selectedBankAccount, 'publishedAt', null)
 
     const [isUpdating, setIsUpdating] = useState(false)
-    const reportVisibleRef = useRef<BankAccountVisibility>(Number(!isNull(accountVisibility)))
+    const [reportVisible, setReportVisible] = useState<BankAccountVisibility>(Number(!isNull(accountVisibility)))
 
     const updateBankAccountReport = BankAccountReport.useUpdate({}, (result) => {
-        reportVisibleRef.current = Number(!isNull(result.publishedAt))
+        setReportVisible(Number(!isNull(result.publishedAt)))
```

Comment:
please move `Number(!isNull(value))` to a helper, it's used multiple times and worsens readability